baseplate/lib/propagator_redditb3.py

```diff
@@ -0,0 +1,158 @@
+import logging
+
+from re import compile as re_compile
+from typing import Any
+from typing import Iterable
+from typing import Optional
+from typing import Set
+
+from opentelemetry import trace
+from opentelemetry.context import Context
+from opentelemetry.propagators.textmap import CarrierT
+from opentelemetry.propagators.textmap import default_getter
+from opentelemetry.propagators.textmap import default_setter
+from opentelemetry.propagators.textmap import Getter
+from opentelemetry.propagators.textmap import Setter
+from opentelemetry.propagators.textmap import TextMapPropagator
+from opentelemetry.trace import format_span_id
+from opentelemetry.trace import format_trace_id
+
+logger = logging.getLogger(__name__)
+
+
+class RedditB3Format(TextMapPropagator):
+    """Propagator for the Reddit B3 HTTP header format."""
+
+    TRACE_ID_KEY = "X-Trace"
+    SPAN_ID_KEY = "X-Span"
+    SAMPLED_KEY = "X-Sampled"
+    FLAGS_KEY = "X-Flags"
+    _SAMPLE_PROPAGATE_VALUES = {"1", "True", "true", "d"}
+    # Although Reddit B3 uses 64bit TraceId's (16 char) we will accept 32 and truncate only when writing client headers.
+    _trace_id_regex = re_compile(r"[\da-fA-F]{16}|[\da-fA-F]{32}")
```

Comment:
Since there's no anchor at the end, I think it will always just select the first 16 chars even for a longer ID?

```python
>>> re.match(r"[\da-fA-F]{16}|[\da-fA-F]{32}", "aaaabbbbccccddddaaaabbbbccccdddd").group(0)
'aaaabbbbccccdddd'
```

It also allows other character lengths as long as they're at least 16. Not sure exactly what the intention of the regex is here but given that the `|[\da-fA-F]{32}` part isn't doing anything, this seems unexpected?

Reply:
I think your correct here, I lifted this from upstream here and modified it for Reddit's format.
https://github.com/open-telemetry/opentelemetry-python/blob/d6321d627cd0fc9345a689a9a37cf4cb9ea44ba0/propagator/opentelemetry-propagator-b3/src/opentelemetry/propagators/b3/__init__.py#L46-L47

Let me dig in a bit more to make sure we get the best behavior.

Reply:
Either way this is incorrect as the baseplate spec uses base10 encoded integers... I'll get a fix together.